### PR TITLE
Add support for spam subject prefix in mailbox settings

### PR DESCRIPTION
**Pull request by stephdl**
does adds support for 

- add  a spam subject prefix in the mailbox settings. 
- remove spam prefix settings from filters > antispam vue page 

refs: https://github.com/NethServer/dev/issues/6898

[Capture vidéo du 2024-03-22 14-39-02.webm](https://github.com/NethServer/ns8-mail/assets/3164851/8ddc357f-7764-4e34-be1a-d0b26ec11b4a)

### CODE REVIEW

File: ui/src/views/settings/SettingsMailboxes.vue [hunk 1]
```diff
@@ -517,6 +535,12 @@ export default {
         enabled: this.sharedseen.enabled,
       };
 
+      // add prefix to spam subject
+      let addPrefixToSpamSubject = {
+        enabled: this.addPrefixToSpamSubject.enabled,
+        prefix: this.addPrefixToSpamSubject.prefix,
```
Comment (andre8244) [inline ui/src/views/settings/SettingsMailboxes.vue:541]: Not sure but it looks like if you enable the prefix toggle, write a prefix in the input box and then disable the toggle, the prefix passed to the API. Is this problematic?

**Comment by stephdl** [comment]
thank @andre8244 for your comment however I tested again my code

- once configured
```
[root@r3-pve ~]# grep -srn 'DOVECOT_SPAM_SUBJECT_PREFIX' /home/mail2/.config/state/
/home/mail2/.config/state/environment:6:DOVECOT_SPAM_SUBJECT_PREFIX=
```

- I enabled and I set something
```
[root@r3-pve ~]# grep -srn 'DOVECOT_SPAM_SUBJECT_PREFIX' /home/mail2/.config/state/
/home/mail2/.config/state/environment:6:DOVECOT_SPAM_SUBJECT_PREFIX=fdgfdg
```
- then I removed the switch without removing the content
```
[root@r3-pve ~]# grep -srn 'DOVECOT_SPAM_SUBJECT_PREFIX' /home/mail2/.config/state/
/home/mail2/.config/state/environment:6:DOVECOT_SPAM_SUBJECT_PREFIX=
```

However do you want I cp your style code that I can read above, I can obviously  change

**Comment by stephdl** [comment]
[Capture vidéo du 2024-03-21 17-57-20.webm](https://github.com/NethServer/ns8-mail/assets/3164851/ab7efa2b-30a7-4e0d-93ce-d0fffbe5d661)

**Comment by stephdl** [comment]
@andre8244 I did like your code style, I need to test back my new code

**Comment by stephdl** [comment]
@andre8244 the code is workable, now I face an issue, the UI allows me for spam_folder and spam_prefix to use any characters because we just validate a string, so `('\"...` are allowed chars, what to do

**Comment by stephdl** [comment]
@DavidePrincipi @andre8244 should be good

**Review by andre8244 — APPROVED** [review]

**Comment by stephdl** [comment]
Notes

for import-module, seems good : https://github.com/NethServer/ns8-mail/blob/0766a7b32ef822d62b3d1f6da22baec60500f777/imageroot/actions/import-module/20import_environment#L22-L23

**Comment by stephdl** [comment]
tested

- migration from NS7
- backup and restoration

**Pull request merged**